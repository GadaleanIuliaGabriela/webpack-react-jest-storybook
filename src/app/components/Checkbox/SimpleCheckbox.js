import React from 'react';
import Checkbox from '@material-ui/core/Checkbox';

const SimpleCheckbox = () => {
  const [state, setState] = React.useState({
    checked: true,
  });

  const handleChange = name => event => {
    setState({checked: false});
  };

  return (
      <Checkbox
        checked={state.checked}
        onClick={handleChange('checked')}
        value="checked"
        inputProps={{
          'aria-label': 'primary checkbox',
        }}
      />
  );
};

export default SimpleCheckbox;