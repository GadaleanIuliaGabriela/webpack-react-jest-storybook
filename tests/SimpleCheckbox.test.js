import React from 'react';
import SimpleCheckbox from '../src/app/components/Checkbox';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<SimpleCheckbox/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('changes the state when clicked', () => {
  const component = renderer.create(<SimpleCheckbox/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});