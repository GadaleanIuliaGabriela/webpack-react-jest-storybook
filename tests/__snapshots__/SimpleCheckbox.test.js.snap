// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the state when clicked 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-8 Mui-checked MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <input
      aria-label="primary checkbox"
      checked={true}
      className="PrivateSwitchBase-input-10"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
      value="checked"
    />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`changes the state when clicked 2`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
    <input
      aria-label="primary checkbox"
      checked={false}
      className="PrivateSwitchBase-input-10"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
      value="checked"
    />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`changes the state when clicked 3`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
    <input
      aria-label="primary checkbox"
      checked={false}
      className="PrivateSwitchBase-input-10"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
      value="checked"
    />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</span>
`;

exports[`renders correctly 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-8 Mui-checked MuiIconButton-colorSecondary"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <input
      aria-label="primary checkbox"
      checked={true}
      className="PrivateSwitchBase-input-10"
      data-indeterminate={false}
      onChange={[Function]}
      type="checkbox"
      value="checked"
    />
  </span>
</span>
`;
